page_forms/base_page.py

```diff
@@ -0,0 +1,106 @@
+import math
+from selenium.common.exceptions import NoSuchElementException
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class BasePage():
+    LOGIN_LINK = (By.CSS_SELECTOR, "#login_link")
+    LOGIN_LINK_INVALID = (By.CSS_SELECTOR, "#login_link_inc")
+    BASKET_LINK = (By.XPATH, "//a[contains(text(), 'View basket')]")
+    USER_ICON = (By.CSS_SELECTOR, ".icon-user")
+
+    def __init__(self, browser, url, timeout=10):
+        self.browser = browser
+        self.url = url
+        self.browser.implicitly_wait(timeout)
+
+    def open(self):
+        self.browser.get(self.url)
+
+    def find_element(self, how, what):
+        try:
+            return self.browser.find_element(how, what)
+        except NoSuchElementException:
+            print(f"Element {what} not found on page")
+            return None
+
+    def find_elements(self, how, what):
+        try:
+            return self.browser.find_elements(how, what)
+        except NoSuchElementException:
+            print(f"No elements found matching {what}")
+            return []
+
+    def get_element_text(self, how, what):
+        element = self.find_element(how, what)
+        if element:
+            return element.text
+        return ""
+
+    def click_element(self, how, what):
+        element = self.find_element(how, what)
+        if element:
+            element.click()
+
+    def enter_text(self, how, what, text):
+        element = self.find_element(how, what)
+        if element:
+            element.clear()
+            element.send_keys(text)
+
+    def handle_alert(self, timeout=10):
```

Comment:
Please try to not hardcode value, especially timeout=10 already duplicated. Let's create constant for default timeout